app/models/generic_event/journey_cancel.rb

```diff
@@ -0,0 +1,11 @@
+class GenericEvent
+  class JourneyCancel < GenericEvent
+    EVENTABLE_TYPES = %w[Journey].freeze
+
+    validates :eventable_type, inclusion: { in: EVENTABLE_TYPES }
```

Comment:
I've looked into drying these validations up and my options are:

1. Use `Tracepoint` to implement behaviour that will add validations via the parent after the `EVENTABLE_TYPES` constant has been defined in the child class (I don't like this - it's non-idiomatic/too complex/unusual and is worse than duplicating in my opinion)
2. Use another level of inheritance for the eventable type (e.g. one for JourneyEvent, PeopleEvent, MoveEvent) which adds eventable-specific validations. Inheritance hierarchies can be very hard to reason about and this implementation actually lacks the flexibility that we might want going forward. I genuinely think this is a can of worms that would end up opening us up to a pattern of very complex STI classes.
3. Duplicate the validation rules in the STI classes for each eventable. This smells a bit, gives us maximum flexibility and is really clear when you read the code.

I've opted for 3 for the reasons noted above. Happy to have a discussion about alternatives, here.

Reply:
I'd personally go with (2) and introduce another layer: `JourneyXXXEvent < JourneyGenericEvent < GenericEvent`, with eventable_type validations at the JourneyGenericEvent level. However, I appreciate hierarchies of classes are not everyone's cup of tea.

Alternatively, how about including the validations as a concern?

Reply:
I'd be happy to use concerns to dry it up. That way we don't get the pitfalls/inflexibility of inheritance.

Reply:
👍